packages/extension-dapp/src/compat/singleSource.ts

```diff
@@ -73,15 +73,13 @@ function injectSingleSource (win: SingleWindow): void {
 // https://github.com/cennznet/singlesource-extension/blob/f7cb35b54e820bf46339f6b88ffede1b8e140de0/react-example/src/App.js#L19
 export default function initSingleSource (): Promise<boolean> {
   return new Promise((resolve): void => {
-    window.addEventListener('load', (): void => {
-      const win = window as Window & SingleWindow;
```

Comment:
I believe we need to drop this anyway, however this on-load used to be important here - before waiting for load, you would get inconsistencies - sometimes it would wok, sometimes it would not.

So this can go in (I really need to drop this, especially since we have the MM version available), but for reference the event listen here was there for a reason.

Reply:
@jacogr is there an efficient way to test reliably this part before merging ?

Reply:
Yes, you can test it against a locally built version of the extension. so 

- disable the one you have
- run yarn build
- add extension from path in Chrome

So if needed, what you will see is inconsistency - sometimes it may sow up, other times it may not show up. (All depending on load order, which is what this construct ensures.) It is a huge PITA, the extension itself also does this stuff on init and now that I mention it... it may just be enough and the above may indeed not be needed now...

Reply:
I don't feel to comfortable changing it, so I'm reverting it back. I'd prefer to have it in another PR

Reply:
I removed the 'load' event because it was preventing the extension from working